libtenzir/test/series_builder.cpp

```diff
@@ -0,0 +1,625 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "tenzir/series_builder.hpp"
+
+#include "tenzir/fwd.hpp"
+
+#include "tenzir/table_slice.hpp"
+#include "tenzir/test/test.hpp"
+
+#include <arrow/api.h>
+
+namespace tenzir {
+
+namespace {
+
+void check(
+  series_builder& b,
+  const std::vector<std::tuple<int64_t, std::string_view, std::string_view>>&
+    ref) {
+  auto total = int64_t{0};
+  for (auto& r : ref) {
+    total += std::get<0>(r);
+  }
+  CHECK_EQUAL(b.length(), total);
+  auto result = b.finish();
+  CHECK_EQUAL(result.size(), ref.size());
+  for (auto i = size_t{0}; i < result.size(); ++i) {
+    auto [len, ty, data] = i < ref.size() ? ref[i] : decltype(ref[i]){};
+    CHECK_EQUAL(result[i].data->length(), len);
+    CHECK_EQUAL(result[i].data->type()->ToString(), ty);
+    auto string = result[i].data->ToString();
```

Comment:
I don't think we should trust that the output of `ToString` will remain stable for future releases. Isn't there something else we can use for these tests?

Reply:
We could print the data as JSON and use that as the baseline.

Reply:
This has been stable for at least two years now, so I don't think we need to address that now.